apps/mobile-wallet/src/components/tokensLists/FtWorth.tsx

```diff
@@ -0,0 +1,49 @@
+import { calculateAmountWorth, CURRENCIES, isFT, TokenId } from '@alephium/shared'
+import { useFetchToken, useFetchTokenPrice } from '@alephium/shared-react'
+import { isNumber } from 'lodash'
+import { useMemo } from 'react'
+import styled from 'styled-components'
+
+import Amount, { AmountProps } from '~/components/Amount'
+import { useAppSelector } from '~/hooks/redux'
+
+interface FtWorthProps extends AmountProps {
+  tokenId: TokenId
+  balance?: bigint
+}
+
+const FtWorth = (props: FtWorthProps) => {
+  const { data: token } = useFetchToken(props.tokenId)
+
+  if (!token || !isFT(token)) return null
+
+  return <FtWorthAmount symbol={token.symbol} decimals={token.decimals} {...props} />
+}
+
+export default FtWorth
+
+interface AddressFtWorthAmountProps extends FtWorthProps {
+  symbol: string
+  decimals: number
+}
+
+const FtWorthAmount = ({ symbol, decimals, balance, ...props }: AddressFtWorthAmountProps) => {
```

Comment:
Linked to my previous comment, this becomes event more confusing 😆 
We should maybe remove the "Amount" part. Also in `calculateAmountWorth` -> `calculateTokenWorth`

Reply:
Yeah ok, naming is indeed hard 😆 So:

- Because those components can accept any "amount", the prop in `FtWorthProps` should probably be `amount` instead of `balance`
- `FtWorthAmount` => `FtAmountWorth`
- We keep `calculateAmountWorth` as it is.